### PR TITLE
[flutter_tools] remove branch migration and standardize constructor style for version interface

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

* Remove the branch migration code - that was 3 years ago or so, its been running long enough.
* Use named parameters for the Version constructor
* Remove version file error handling that is superseded by error_handling_io.dart
* Remove isMaster , which is only used to avoid showing the welcomeToFlutter check on master,

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart [hunk 1]
```diff
@@ -385,7 +385,7 @@ void main() {
 }
 
 class FakeUpgradeCommandRunner extends UpgradeCommandRunner {
-  bool willHaveUncomittedChanges = false;
+  bool willHaveUncommittedChanges = false;
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart:388]: typo

File: packages/flutter_tools/lib/src/runner/flutter_command_runner.dart [hunk 2]
```diff
@@ -242,13 +242,7 @@ class FlutterCommandRunner extends CommandRunner<void> {
           globals.flutterUsage.suppressAnalytics = true;
         }
 
-        try {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/runner/flutter_command_runner.dart:245]: I think this is now entirely covered by error_handling_io.

File: packages/flutter_tools/lib/src/version.dart [hunk 3]
```diff
@@ -16,17 +16,6 @@ import 'globals.dart' as globals;
 /// The flutter GitHub repository.
 String get _flutterGit => globals.platform.environment['FLUTTER_GIT_URL'] ?? 'https://github.com/flutter/flutter.git';
 
-/// This maps old branch names to the names of branches that replaced them.
-///
-/// For example, in early 2018 we changed from having an "alpha" branch to
-/// having a "dev" branch, so anyone using "alpha" now gets transitioned to
-/// "dev".
-const Map<String, String> kObsoleteBranches = <String, String>{
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/version.dart:24]: 3 years is long enough

File: packages/flutter_tools/lib/src/reporting/usage.dart [hunk 4]
```diff
@@ -423,8 +423,7 @@ class _DefaultUsage implements Usage {
         isFirstRun ||
         // Display the welcome message if we are not on master, and if the
         // persistent tool state instructs that we should.
-        (!globals.flutterVersion.isMaster &&
-        (globals.persistentToolState.redisplayWelcomeMessage ?? true))) {
+        (globals.persistentToolState.redisplayWelcomeMessage ?? true)) {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/reporting/usage.dart:426]: why did you remove the master check?

File: packages/flutter_tools/lib/src/version.dart (hunk 3 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/version.dart:24]: RIP alpha channel

File: packages/flutter_tools/lib/src/commands/downgrade.dart [hunk 5]
```diff
@@ -87,7 +87,7 @@ class DowngradeCommand extends FlutterCommand {
     String workingDirectory = Cache.flutterRoot;
     if (argResults.wasParsed('working-directory')) {
       workingDirectory = stringArg('working-directory');
-      _flutterVersion = FlutterVersion(const SystemClock(), workingDirectory);
+      _flutterVersion = FlutterVersion(clock: const SystemClock(), workingDirectory: workingDirectory);
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/downgrade.dart:90]: you can skip passing the `clock` param, since `SystemClock()` is the default value. This was only explicitly added because it was positional.

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/commands/downgrade.dart:90]: although, explicitly showing deps is good, too...

File: packages/flutter_tools/lib/src/reporting/usage.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/reporting/usage.dart:426]: This is the only place we're doing this - I think practically there isn't really a need to hide master since it should already be hidden on bots

File: packages/flutter_tools/lib/src/commands/downgrade.dart (hunk 5 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/downgrade.dart:90]: Ahh, good point

File: packages/flutter_tools/lib/src/reporting/usage.dart (hunk 4 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/reporting/usage.dart:426]: I thought this was an intentional usability improvement for devs on the master channel. It was added here: https://github.com/flutter/flutter/pull/43217/files. That having been said, I don't have a strong preference either way, before this change I just ignored the message.

**Review by christopherfujino — APPROVED** [review]
LGTM
